src/new-client/src/app/site/[site]/details/edit-information-for-citizens/add-information-for-citizens-form.tsx

```diff
@@ -48,16 +52,57 @@ const AddInformationForCitizensForm = ({
     replace(`/site/${site}/details`);
   };
 
+  const isValidTextInput = (text: string): boolean => {
+    const urlRegex = new RegExp(
+      '([a-zA-Z0-9]+://)?([a-zA-Z0-9_]+:[a-zA-Z0-9_]+@)?([a-zA-Z0-9.-]+\\.[A-Za-z]{2,4})(:[0-9]+)?([^ ])+',
```

Comment:
This might be a good opportunity to introduce [schema driven validation](https://www.react-hook-form.com/get-started/#SchemaValidation) so we could probably do something like `yup().string().not().email()` and save having to write a regex ourselves. 
Admittedly it's scope creep, it's just something I've been meaning to get around to doing at some point so thought I'd mention it :)

Reply:
Ooh actually I've just remember we have a `constants.tsx` file that already has an `EMAIL_REGEX` in it. On the one hand I'd rather scrap this file and replace it with schema validation at some point, but on the other hand if we want to hold off for now and introduce schemas at a later date we should probably avoid the repetition and use this here. We could add a second variable like `SPECIAL_CHARACTERS_REGEX` or something to handle the other cases, whatever suits :)

Reply:
Didn't realise that file existed, I'll stick these two in there and we could always add the schema driven validation at a later date. It's also a URL regex not another email one so there will be 3 in there now :)